### PR TITLE
Admin inning start date validation

### PR DESCRIPTION
**Pull request by AlejandroLopez1992**
__X_ Wrote Tests _X__ Implemented _X__ Reviewed

## Necessary checkmarks:

- [x] All Tests are Passing in all environments

- [x] The code will run locally

## Type of change

- [x] New feature
- [ ] Bug Fix
- [ ] Refactor

## Description of Change:

    description: Add validations for date_within_allowed_range and update testing to handle validation.

## Requested feedback:

    I'd like feedback on, possibly refactoring the date_within_allowed_range method or optimizations. We are also wondering if we should put additional validation disallowing the edit of the current inning start_date.

## Check the correct boxes

- [x] This broke nothing
- [x] This broke some stuff
- [x] This broke everything

## Testing Changes

- [ ] No Tests have been changed
- [x] Some Tests have been changed
- [ ] All of the Tests have been changed(Please describe what in the world happened)

## Checklist:

- [x] My code has no unused/commented out code
- [x] My code has no binding.pry's
- [x] I have reviewed my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have fully tested my code

(For Fun!)Please Include a link to a gif of your feelings about this branch

Link: 
image

### CODE REVIEW

File: app/models/inning.rb [hunk 1]
```diff
@@ -1,9 +1,18 @@
...
   validates_presence_of :name, :start_date
+  validate :date_within_allowed_range
 
   has_many :turing_modules, dependent: :destroy
   has_many :students, through: :turing_modules
 
+  def date_within_allowed_range
+    if Inning.none?
+      # Execute as normal without triggering validation error
+    else current_inning_date = Inning.where(current: true).first.start_date
```
Comment (BrianZanti) [inline app/models/inning.rb]: A little syntax tip: `.where.first` can be replaced by `find_by`

File: app/models/inning.rb [hunk 2]
```diff
@@ -1,9 +1,18 @@
 class Inning < ApplicationRecord
   validates_presence_of :name, :start_date
+  validate :date_within_allowed_range
 
   has_many :turing_modules, dependent: :destroy
   has_many :students, through: :turing_modules
 
+  def date_within_allowed_range
+    if Inning.none?
+      # Execute as normal without triggering validation error
```
Comment (BrianZanti) [inline app/models/inning.rb:10]: This is a helpful comment

File: app/models/inning.rb [hunk 3]
```diff
@@ -49,4 +59,10 @@ def create_turing_modules
       turing_modules.create!(program: 'BE', module_number: i + 1)
     end
   end
+
+  def date_validation(current_inning_date)
+    return errors.add(:start_date, "Can't be blank") if start_date.nil?
```
Comment (BrianZanti) [inline app/models/inning.rb]: This would be a good place to use `.blank?` instead of `.nil` since blank will also catch an empty string.

**Review by BrianZanti — COMMENTED** [review]
Great work! There's one minor change in here I'd like you to make.

File: app/models/inning.rb (hunk 1 above)
Comment (AlejandroLopez1992) [reply app/models/inning.rb]: Thank you, switched this to use 'find_by_current(true)' to improve readability and syntax

File: app/models/inning.rb (hunk 3 above)
Comment (AlejandroLopez1992) [reply app/models/inning.rb]: awesome TY~! Updated to use .blank?

**Review by BrianZanti — APPROVED** [review]

**Pull request merged**